### PR TITLE
Dependency-Injecting DAOs and dependents

### LINKED ISSUE #3059: Towards Play 2.6: Use Dependency Injection for DAOs + Services
part of #1622

### PR DESCRIPTION
**Pull request by fm3**
This PR is a big step towards upgrading Play. It
 - significantly reduces the (deprecated) usage of the global state of the framework `Play.current`
 - reduces the usage of `GlobalSettings`
 - …by creating most Services and DAOs as dependency-injected classes
 - it removes circular dependencies amongst these

 -  note that not all of the newly injected classes are initialized `asEagerSingleton` – we might add that as a further optimization, but I don’t want to make this PR any bigger right now.
 - Nothing can be injected into case classes, so they can’t provide much functionality anymore, stuff like `annotation.task` has been moved to `annotationService.taskFor(annotation)`. Same with `publicWrites` methods
 - since silhouette is now injected into the controllers, the implicit conversions from SecuredRequest to DBAccessContext can’t be in a trait anymore, which leads to a bit of code duplication, but I think we can live with that for now
 - the resolution of cyclic dependencies brought some refactoring, for example we no longer need the explicit conversion `securedRequestToUserAwareRequest` (we pass on the Option[User] rather than the request)
 - there are still eight occurances of Play.current, which block advancing to 2.6 together with the streaming stuff (and a possibly required silhouette upgrade). We will adress those in future PRs

### TODO
 - [x] remove currently commented-out `ProvidesUnauthorizedSessionData` (I believe it is unused or fully replaced by silhouette, since the tests run through and manual testing didn’t show changed behavior, I don’t think we need it anymore)
 - [x] ensureTeamAdministration should be replaced by `assertTrue(userService.isTeamManagerOrAdminOf)` – the `assertTrue` is missing from some places
 - [x] `annotationService.zippedFourLists` should be renamed

### Steps to test:
 - stuff the CI doesn’t cover:
 - reload after scala code changes should close the db handle + open a new one (check console output)
 - Mailer should still work
 - cleanup services should still work (try reducing the interval for testing)

### Issues:
- fixes #3059 
- contributes to #3102 
(which are both part of #1622 )

------
- ~~[ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
- ~~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- ~~[ ] Updated [documentation](../blob/master/docs) if applicable~~
~~- [ ] Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

File: app/oxalis/security/CredentialsProvider.scala [hunk 1]
```diff
@@ -27,7 +27,7 @@ import scala.concurrent.{ExecutionContext, Future}
   * //@param passwordHasherList List of password hasher supported by the application.
   * @param executionContext The execution context to handle the asynchronous operations.
   */
-class CredentialsProvider @Inject() (passwordHasher: PasswordHasher)(implicit val executionContext: ExecutionContext)
+class CredentialsProvider @Inject()(userService: UserService) (passwordHasher: PasswordHasher)(implicit val executionContext: ExecutionContext)
```
Comment (jstriebel) [inline app/oxalis/security/CredentialsProvider.scala]: How does the syntax work here? So many constructor argument braces…

File: app/controllers/InitialDataController.scala [hunk 2]
```diff
@@ -79,31 +88,31 @@ Samplecountry
...
+          _ <- userTeamRolesDAO.insertTeamMembership(defaultUser._id, TeamMembership(organizationTeam._id, true))
           _ = logger.info("Inserted default user scmboy")
         } yield ()
     }.toFox
   }
 
   def insertToken = {
-    val expiryTime = Play.configuration.underlying.getDuration("silhouette.tokenAuthenticator.authenticatorExpiry").toMillis
-    TokenDAO.findOneByLoginInfo("credentials", defaultUserEmail, TokenType.Authentication).futureBox.flatMap {
+    val expiryTime = conf.Silhouette.TokenAuthenticator.authenticatorExpiry.toMillis
```
Comment (jstriebel) [inline app/controllers/InitialDataController.scala]: :+1:

File: app/controllers/Authentication.scala [hunk 3]
```diff
@@ -97,21 +93,32 @@ object AuthForms {
...
+                                organizationDAO: OrganizationDAO,
+                                userDAO: UserDAO,
+                                sil: WebknossosSilhouette,
+                                val messagesApi: MessagesApi
+)
   extends Controller
-    with ProvidesUnauthorizedSessionData
     with FoxImplicits {
 
+  implicit def userAwareRequestToDBAccess(implicit request: sil.UserAwareRequest[_]) = DBAccessContext(request.identity)
```
[the remark this answers is not in the document]
Comment (fm3) [reply app/controllers/Authentication.scala]: I spent quite some time trying to find a generic solution like what you’re suggesting, but I couldn’t make it work. If you find a way, I would be more than happy to change it to that later. But I think that should not block this PR

**Comment by fm3** [comment]
Thanks for the review @jstriebel ! Could I persuade you to hit _approve_ despite the non-generic Request conversions? I created a separate issue for that: #3151

**Pull request merged**